bigbluebutton-html5/imports/ui/components/lock-viewers/notify/component.jsx

```diff
@@ -0,0 +1,64 @@
+import { Component } from 'react';
+import { defineMessages, injectIntl } from 'react-intl';
+import { notify } from '/imports/ui/services/notification';
+import _ from 'lodash';
+
+const intlMessages = defineMessages({
+  disableCam: {
+    id: 'app.userList.userOptions.disableCam',
+    description: 'label to disable cam notification',
+  },
+  disableMic: {
+    id: 'app.userList.userOptions.disableMic',
+    description: 'label to disable mic notification',
+  },
+  disablePrivChat: {
+    id: 'app.userList.userOptions.disablePrivChat',
+    description: 'label to disable private chat notification',
+  },
+  disablePubChat: {
+    id: 'app.userList.userOptions.disablePubChat',
+    description: 'label to disable private chat notification',
+  },
+  onlyModeratorWebcam: {
+    id: 'app.userList.userOptions.webcamsOnlyForModerator',
+    description: 'label to disable all webcams except for the moderators cam',
+  },
+});
+
+class LockViewersNotifyComponent extends Component {
+  componentDidMount() {}
+
+  componentDidUpdate(prevProps) {
+    const {
+      lockSettings,
+      intl,
+      webcamsOnlyForModerator,
+    } = this.props;
+
+    const {
+      lockSettings: prevLockSettings,
+      webcamsOnlyForModerator: prevWebcamsOnlyForModerator,
+    } = prevProps;
+
+    if (!_.isEqual(lockSettings, prevLockSettings)) {
+      const filteredSettings = Object.keys(lockSettings)
+        .filter(key => prevLockSettings[key] !== lockSettings[key]
+          && lockSettings[key]
+          && key !== 'setBy');
+      filteredSettings.forEach((key) => {
+        notify(intl.formatMessage(intlMessages[key]), 'info', 'lock');
+      });
+    }
+    if (webcamsOnlyForModerator !== prevWebcamsOnlyForModerator
```

Comment:
I would rearrange it to first check for `webcamsOnlyForModerator` so we can exit early if it's null

Reply:
It could also just be `if (webcamsOnlyForModerator && !prevWebcamsOnlyForModerator) {}`

Reply:
Looks like the same change should really be done in the filter also.

Reply:
If statement rearranged.